### PR TITLE
handheld Experimenters give more helpful feedback

### PR DESCRIPTION
**Pull request by Bilbo367**
## About The Pull Request

Handheld experimenters now will tell you if the machine parts are not high enough tier, not the exact needed tier, if scanned mechs are missing equipment in arms, and if scanned mech is not hand-crafted.

## Why It's Good For The Game

Helps players (me) understand why I can't scan the kiosk for the experiment (I didn't read).

## Changelog

:cl:
qol: some experiments now give you more helpful hints at why your failing
code: changes around some experiment code to make it more understandable
/:cl:

### CODE REVIEW

File: code/modules/experisci/experiment/experiments.dm [hunk 1]
```diff
@@ -323,32 +323,12 @@
...
-/datum/experiment/scanning/random/mecha_equipped_scan/final_contributing_index_checks(atom/target, typepath)
-	var/obj/vehicle/sealed/mecha/stompy = target
-	if(!istype(stompy))
-		return FALSE //Not a mech
-	if(!HAS_TRAIT(stompy,TRAIT_MECHA_CREATED_NORMALLY))
-		return FALSE //Not hand-crafted
-	if(!(stompy.equip_by_category[MECHA_L_ARM] && stompy.equip_by_category[MECHA_R_ARM]))
-		return FALSE //Both arms need be filled
-	return ..()
-
```
Comment (Bilbo367) [inline code/modules/experisci/experiment/experiments.dm:351]: These are moved. Only experiment defines are in this file, although few are not.

File: code/modules/experisci/experiment/types/random_scanning.dm [hunk 2]
```diff
@@ -19,3 +19,25 @@
...
+
+/datum/experiment/scanning/random/mecha_equipped_scan/final_contributing_index_checks(datum/component/experiment_handler/experiment_handler, atom/target, typepath)
+	var/obj/vehicle/sealed/mecha/stompy = target
+	if(!istype(stompy))
+		return FALSE //Not a mech
+	if(!HAS_TRAIT(stompy,TRAIT_MECHA_CREATED_NORMALLY))
+		experiment_handler.announce_message("Scanned mech was not made by crew. Their is nothing to learn here.")
+		return FALSE //Not hand-crafted
+	if(!(stompy.equip_by_category[MECHA_L_ARM] && stompy.equip_by_category[MECHA_R_ARM]))
+		experiment_handler.announce_message("Scanned mech is missing equipment on their arms to proceed this experiment.")
```
Comment (Bilbo367) [inline code/modules/experisci/experiment/types/random_scanning.dm:41]: Please tell me if anyone wants some other logics announce why the experiment is failing.

Comment (Sadboysuss) [reply code/modules/experisci/experiment/types/random_scanning.dm:41]: also: "Their is nothing to learn here", minor spelling nitpick

Comment (Bilbo367) [reply code/modules/experisci/experiment/types/random_scanning.dm:41]: Whats the spelling nitpick? your saying what is already in the code.

Comment (00-Steven) [reply code/modules/experisci/experiment/types/random_scanning.dm:41]: That would be "There is nothing to learn here."
Which to use when out of their/they're/there can be hard shit to learn, so I getcha

**Pull request merged**